packages/google_sign_in/tizen/src/secure_storage.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_PLUGIN_SECURE_STORAGE_H_
+#define FLUTTER_PLUGIN_SECURE_STORAGE_H_
+
+#include <ckmc/ckmc-type.h>
+
+#include <optional>
+#include <string>
+#include <variant>
+#include <vector>
+
+class SecureStorage {
+ public:
+  SecureStorage();
+
+  virtual ~SecureStorage();
+
+  void CreateKey();
+
+  bool HasKey() const;
+
+  void Destroy();
+
+  void SaveData(const std::string& name, std::vector<uint8_t> data,
```

Comment:
How about using `const` and `&` for vectors?

Reply:
The first parameter `data` in `ckmc_buffer_new` has a cv-qualifier of non-const. I could use `const_cast`, would that be better?

Reply:
You can pass the `data` by reference and explicitly copy it to a local variable. The current `SaveData` results in a double implicit copy operation.

```patch
@@ -127,7 +128,8 @@ std::vector<uint8_t> SecureStorage::EncryptData(
   ckmc_param_list_set_buffer(params_, CKMC_PARAM_ED_IV, buffer);
   ckmc_buffer_free(buffer);

-  ckmc_buffer_new(data.data(), data.size(), &buffer);
+  std::vector<uint8_t> plain(data);
+  ckmc_buffer_new(plain.data(), plain.size(), &buffer);
   ckmc_encrypt_data(params_, kAesKey, nullptr, *buffer, &encrypted_buffer);
   ckmc_buffer_free(buffer);
```